apps/bluetooth/js/modules/pair_manager.js

```diff
@@ -38,21 +62,176 @@ define(function(require) {
           this.showPendingPairing(event.settingValue);
       }.bind(this));
 
-      // Observe Bluetooth ondisabled event from hardware side.
+      // Observe Bluetooth state from hardware side.
       // Then, close pairing dialog immediately.
-      navigator.mozBluetooth.ondisabled = this.onBluetoothDisabled.bind(this);
+      BtContext.observe('state', (state) => {
```

Comment:
It makes more sense to observe 'enabled' as `onBleteoothDisabled` is supposed to be called on stable state.